test/lib/next-modes/next-dev.ts

```diff
@@ -152,55 +152,13 @@ export class NextDevInstance extends NextInstance {
 
   public override async patchFile(
     filename: string,
-    content: string | ((contents: string) => string),
+    content: string | ((content: string) => string),
     runWithTempContent?: (context: { newFile: boolean }) => Promise<void>
   ) {
-    const isServerRunning = this.childProcess && !this.isStopping
-    const cliOutputLength = this.cliOutput.length
-
-    if (isServerRunning) {
-      await this.handleDevWatchDelayBeforeChange(filename)
-    }
-
-    const waitForChanges = async ({ newFile }: { newFile: boolean }) => {
-      if (isServerRunning) {
-        if (newFile) {
-          await this.handleDevWatchDelayAfterChange(filename)
-        } else if (filename.startsWith('next.config')) {
-          await retry(async () => {
-            const cliOutput = this.cliOutput.slice(cliOutputLength)
-
-            if (!this.serverReadyPattern.test(cliOutput)) {
-              throw new Error('Server has not finished restarting.')
-            }
-          })
```

Comment:
We don't need/want this anymore?

Reply:
> It's not needed in general based on the test results.

I'm not convinced that this is enough proof that we don't need it anymore. It solved a real issue when we introduced it in #67601.

> If it is needed in specific tests, we should add the delay over there.

I think that adding an arbitrary delay is worse than waiting for a specific output, wouldn't you agree?

Reply:
There was already 5 second delay for all patchFile (unless the output is detected, which rarely happens). What the PR does is to allow individual tests to customize the delay. But agree that eventually there should be no delay.

Reply:
This comment is referring to the `serverCompiledPattern` though, right? Whereas this thread is regarding the `serverReadyPattern`. Waiting for the latter after a next.config change in the dev server basically has no downside and is always expected.

Reply:
If that detection is reliable, I can add it back in!

Reply:
@unstubbable Thanks for this feedback - I added back the case for `serverReadyPattern` -> https://github.com/vercel/next.js/pull/72439/commits/b23b85854a45e8aec1d8160ccb817a9cbb8dac3b